### PR TITLE
issue 496.4: move StorageEngine-specific flag out of the genric API and closer to the storage engine

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
https://github.com/arangodb/enterprise/pull/167
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1879/

File: arangod/VocBase/LogicalCollection.cpp [hunk 1]
```diff
@@ -616,7 +627,6 @@ arangodb::Result LogicalCollection::appendVelocyPack(
     // with 'forPersistence' added by LogicalDataSource::toVelocyPack
     // FIXME TODO is this needed in !forPersistence???
     result.add(StaticStrings::DataSourceDeleted, VPackValue(deleted()));
-    result.add(StaticStrings::DataSourceGuid, VPackValue(guid()));
```
Comment (graetzer) [inline arangod/VocBase/LogicalCollection.cpp]: Why did you remove that ?

Comment (vasiliy-arangodb) [reply arangod/VocBase/LogicalCollection.cpp]: because previously DataSourceGuid was output by LogicalDataSource if forPersistence and by LogicalCollection if !forPersistence, now it's always output by LogicalDataSource

**Comment by vasiliy-arangodb** [comment]
replaced by https://github.com/arangodb/arangodb/pull/7212
